server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -249,6 +264,7 @@ public static Version fromString(String version) {
         this.revision = (byte) ((id / 100) % 100);
         this.build = (byte) (id % 100);
         this.luceneVersion = Objects.requireNonNull(luceneVersion);
+        this.toString = major + "." + minor + "." + revision;
```

Comment:
No need to re-create the same string over and over for the same version constants (obviously in some cases this is wasted memory when we work with version constants that aren't cached but that's rare and the overhead is trivial anyway).